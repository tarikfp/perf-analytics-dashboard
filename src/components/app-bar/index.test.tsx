import { render } from '@testing-library/react';
import * as React from 'react';
import MenuAppBar from '.';
import {
  createWithThemeContext,
  wrapWithThemeContext,
} from '../../utils/test-utils';

test('app bar renders correctly', () => {
  const tree = createWithThemeContext(<MenuAppBar />);
  expect(tree).toMatchSnapshot();
});

test('app bar header text renders correctly', () => {
  const { getByText } = render(wrapWithThemeContext(<MenuAppBar />));
  expect(getByText('Performance analytics dashboard')).toBeTruthy();
});

test('app bar icon button renders correctly', () => {
  const { getByRole } = render(wrapWithThemeContext(<MenuAppBar />));
  expect(getByRole('button')).toBeTruthy();
});
