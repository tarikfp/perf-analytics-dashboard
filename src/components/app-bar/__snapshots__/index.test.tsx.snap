// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app bar renders correctly 1`] = `
<div
  className="MuiBox-root css-i9gxme"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1622sjm-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
    >
      <div
        className="MuiTypography-root MuiTypography-h6 css-43ikqx-MuiTypography-root"
      >
        Performance analytics dashboard
      </div>
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          color="#29434e"
          data-testid="Brightness4Icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6c3.31 0 6 2.69 6 6s-2.69 6-6 6z"
          />
        </svg>
      </button>
    </div>
  </header>
</div>
`;
